feat: <ContentTitles /> and other new components

conrad-vanl opened the pull request:
## DESCRIPTION

### What does this PR do, or why is it needed?

- Adds `<ContentTitles />` component
- Adds`<SocialBar />` component (to replace ActionBar and other usages of like/sharing buttons)
- Adds experimental Reanimated-based `<StretchyView />` component
- Adds `useLike` and `useShare` hooks
- Updates `NodeSingleConnected`, `ContentNodeConnected`, `DefaultCard`, and `FeaturedCard` to use new components, which starts the process of migrating some of our most commonly used card components to the designs we've been using for almost a year in Figma.

### How do I test this PR?

Stories:

- `NodeSingleConnected`
- `DefaultCard`
- `FeaturedCard`

conrad-vanl commented:
@vinnyjth yeah changes to files in the `theme/` directory could be moved out to a separate PR.

I do want to start migrating theme to fall in line with UI updates that are in Figma, so hopefully that PR can still be merged and not just put off.

conrad-vanl commented:
I could also try and .... adjust the theme colors some so that there's not a ton of snap changes while still updating theme

conrad-vanl commented:
Note: Going to close this in favor of a different approach

conrad-vanl commented:
@redreceipt this one kind of got away from me, but I think it's a net positive change:

- Introduces new components that match in both naming and props to our Figma UI-Kit
- Uses those components across Cards and Node views
- Reduces overall code footprint some 

This also leaves a couple of components that we can deprecate and remove:

- ContentCard (already deprecated but we can remove it at any time I think)
- LikeButtonConnected (should no longer be used on templates)
- ShareButtonConnected (should no longer be used on templates)

This is ready for review.

conrad-vanl commented:
Also throwing @rkdavidson on here as a potential reviewer

redreceipt commented on packages/apollos-ui-connected/src/LikeButtonConnected/useLike.js:
```diff
@@ -0,0 +1,68 @@
+import GET_LIKED_NODE from './getLikedNode';
+import updateLikedContent from './updateLikedContent';
+
+const useLiked = (nodeId) => {
```
```suggestion
const useLike = (nodeId) => {
```

redreceipt commented on packages/apollos-ui-connected/src/LikeButtonConnected/useLike.js:
```diff
@@ -0,0 +1,68 @@
+  return [isLiked, like];
+};
+
+export default useLiked;
```
```suggestion
export default useLike;
```

redreceipt commented:
this is awesome. couple of things:

the actionsheet is now showing up over tabs

<img width="562" alt="Screen Shot 2021-06-30 at 10 39 54 AM" src="https://user-images.githubusercontent.com/2659478/123980499-96e3e300-d98f-11eb-959f-1e82f2816c52.png">

redreceipt commented:
content is now a bit washed out because I'm guessing there's no gradient

<img width="562" alt="Screen Shot 2021-06-30 at 10 41 00 AM" src="https://user-images.githubusercontent.com/2659478/123980645-b24eee00-d98f-11eb-92a0-f388eec27d3e.png">

redreceipt commented:
an accompanying templates PR to remove the like and share bar should come with this so we have an option for people to upgrade too if they want to use these new components

<img width="562" alt="Screen Shot 2021-06-30 at 10 42 45 AM" src="https://user-images.githubusercontent.com/2659478/123981061-08239600-d990-11eb-949e-00eb302b7f46.png">

conrad-vanl commented:
> the actionsheet is now showing up over tabs

This PR didn't change this.... at least to my knowledge.

> content is now a bit washed out because I'm guessing there's no gradient

Hm....yeah. I'll look into it. I know highlight cards we are either going to drop or change design of. I didn't mean to change those in this PR, so I'll dig in and see what I did.

BTW - how did you run the templates app while pointing at this branch?

redreceipt commented:
> BTW - how did you run the templates app while pointing at this branch?

`yarn app`

redreceipt commented:
> > the actionsheet is now showing up over tabs
> 
> This PR didn't change this.... at least to my knowledge.

this may have been a fluke, force closing out, I can't reproduce

redreceipt commented:
also, drop some `console.warns` in the deprecated components so we can remove at some point and say we warned people

redreceipt commented on packages/apollos-ui-kit/src/BackgroundView/index.js:
```diff
@@ -1,17 +1,38 @@
+  const theme = useTheme();
+  const backgroundColor =
+    theme?.colors?.background[material] || theme?.colors?.background?.screen;
+  const style = useMemo(
```
there a reason you wouldn't just use a style sheet here? if someone wanted to change the `View` component, we should prob make it a `Component` prop and do `named("ui-kit.BackgroundView")(BackgroundView) to match our convention. 

then you don't need that `PropTypes.any` thing

conrad-vanl replied:
Not using a style sheet since the styles are based off of props. Not passing an object into a style property because that changes every render, `useMemo` is the "correct" way to do that now.

You're right, the alternative is to use `styled`. Then, the component would become:

```jsx
const BackgroundView = styled(({ theme, material, flexed }) => ({
  backgroundColor: theme?.colors?.background[material] || theme?.colors?.background?.screen,
  flex: flexed ? 1 : 0,
  height: flexed ? '100%' : undefined,
}, 'ui-kit.BackgroundView')(View);
```

I'm guessing that would be preferred. fixing now...

conrad-vanl replied:
Did this. committed.

conrad-vanl commented:
Alright @redreceipt check it out now...

I updated `DefaultCard` some more so that it supports some of the overlay image effects (so that we don't have to pass theme into cards for custom colored cards). They'll look like this now:

![Screen Shot 2021-06-30 at 12 50 22 PM](https://user-images.githubusercontent.com/103927/124008154-bdab1500-d9a1-11eb-856f-6abc01ea5aba.png)

Also, `DefaultCard` supports all of the features of `FeaturedCard` now, so I made `FeaturedCard` component really simple by just having it render a `<DefaultCard featured {...props} />`

I added `console.warn`'s to `HighlightCard`s and fixed the overlay effect on them.

Lastly, I changed our card mappers to no longer return `HighlightCard`s

redreceipt commented on packages/apollos-ui-connected/src/ContentCardConnected/ContentCardComponentMapper/cardMapper.js:
```diff
@@ -1,19 +1,9 @@
-import { DefaultCard, HighlightCard } from '@apollosproject/ui-kit';
+import { DefaultCard } from '@apollosproject/ui-kit';
 
 const cardMapper = (props) => {
```
in this case we don't need a mapper, just a component they can override

conrad-vanl replied:
Mapper needs to stay for backwards compatibility I think

conrad-vanl replied:
Component override is already supported, so I think we just need to throw a.... well I don't want to throw a `console.warn` cuz we'll get blasted with that since it's still ran, but we shouldn't remove it altogether yet? Just don't encourage its use anymore.....

redreceipt replied:
I'm suggested we _don't_ use it anywhere and then warn people it's deprecated

conrad-vanl replied:
It's an override though....how would we warn people it's deprecated?

conrad-vanl replied:
Like if we remove it from our code, that overrides won't work anymore

redreceipt replied:
right before the return statement

conrad-vanl replied:
So leave it in the codebase, just add `console.warn`?

That means the console.warn will get triggered every time a `ContentCardConnected` component is rendered

conrad-vanl replied:
Let me think of a solution.... @redreceipt can you link me to a repo where this override is used somewhere?

redreceipt replied:
right, `ContentCardConnected` doesn't need mapper by default, just `DefaultCard`

redreceipt commented on packages/apollos-ui-connected/src/ContentCardConnected/ContentCardComponentMapper/contentCardComponentMapper.js:
```diff
@@ -1,7 +1,6 @@
-import { DefaultCard, HighlightCard } from '@apollosproject/ui-kit';
+import { DefaultCard } from '@apollosproject/ui-kit';
 
 const contentCardComponentMapper = (props) => {
```
same comment as above. don't need mappers at this point which is awesome because it's unconventional and confusing

redreceipt replied:
actually, this is already a dead component

conrad-vanl commented on packages/apollos-ui-kit/src/FeaturedCard/index.js:
```diff
@@ -1,226 +1,9 @@
-};
-
-FeaturedCard.displayName = 'FeaturedCard';
+const FeaturedCard = (props) => <DefaultCard featured {...props} />;
```
I think there's value to keeping it for now - the way we use it today is as a component override - `CardComponent={FeaturedCard}` is a lot better then say....

```
CardComponent={(props) => <DefaultCard featured {...props})}
```

redreceipt replied:
I agree it's better, but that's our system. We tell people that's how you do it. And not removing it just leaves us maintaining 9 lines of dead code.

Chances are they won't override anyway, it's used internally through features.

redreceipt replied:
fair

redreceipt commented:
> Alright @redreceipt check it out now...
> 
> I updated `DefaultCard` some more so that it supports some of the overlay image effects (so that we don't have to pass theme into cards for custom colored cards). They'll look like this now:
> 
> ![Screen Shot 2021-06-30 at 12 50 22 PM](https://user-images.githubusercontent.com/103927/124008154-bdab1500-d9a1-11eb-856f-6abc01ea5aba.png)
> 
> Also, `DefaultCard` supports all of the features of `FeaturedCard` now, so I made `FeaturedCard` component really simple by just having it render a `<DefaultCard featured {...props} />`
> 
> I added `console.warn`'s to `HighlightCard`s and fixed the overlay effect on them.
> 
> Lastly, I changed our card mappers to no longer return `HighlightCard`s

this is freaking cool

redreceipt reviewed (dismissed):
looks good on the sim. needs an accompanying templates PR to pull out the old like/share bar

conrad-vanl commented:
Okay @redreceipt this should be g2g now, added a pr on templates https://github.com/ApollosProject/apollos-templates/pull/418

redreceipt commented:
@conrad-vanl here's some content with no picture

<img width="562" alt="Screen Shot 2021-06-30 at 3 06 34 PM" src="https://user-images.githubusercontent.com/2659478/124017509-cb1dca80-d9b4-11eb-8046-f566edabadac.png">

conrad-vanl commented:
> @conrad-vanl here's some content with no picture

What are you trying to show with this? Cards with no images show up pretty much like that on core now, I don't think that's a regression

conrad-vanl commented:
@redreceipt anything else you want to see on this before approving?

redreceipt commented:
Only "breaking" thing is you can't click the share button when there's no image. Good if you are?

conrad-vanl commented:
Oh, okay I'll fix that I see now.

Also, fixing the contentcardmapper deprecation thing forgot about that. You can approve if you want and I'll get those two changes pushed up and merged :)
